app/src/main/java/com/g/tragosapp/ui/cocktaildetails/CocktailsDetailFragment.kt

```diff
@@ -46,11 +52,64 @@ class CocktailsDetailFragment : Fragment() {
         binding.cocktailTitle.text = cocktail.name
         binding.cocktailDesc.text = cocktail.description
 
-        binding.btnSaveCocktail.setOnClickListener {
-            viewModel.saveCocktail(FavoritesEntity(cocktail.cocktailId,cocktail.image,cocktail.name,cocktail.description,cocktail.hasAlcohol))
-            Toast.makeText(requireContext(), "Cocktail saved to favorites", Toast.LENGTH_SHORT)
-                .show()
-        }
+        viewModel.getFavoriteCocktails().observe(viewLifecycleOwner, Observer { result ->
+            when (result) {
+                is Resource.Loading -> {
+                }
+                is Resource.Success -> {
+                    when (result.data.contains(cocktail)) {
+                        true -> {
+                            binding.btnSaveCocktail.setOnClickListener {
+                                viewModel.deleteCocktail(
+                                    FavoritesEntity(
+                                        cocktail.cocktailId,
+                                        cocktail.image,
+                                        cocktail.name,
+                                        cocktail.description,
+                                        cocktail.hasAlcohol
+                                    )
```

Comment:
Save and delete has this code duplicated, we can extract this to prevent the duplication

Reply:
Are you dumb? YOU HAVE HALF OF THE PROJECT IS A DUPLICATION!!! want to help people but they are talking stupid and do not appreciate .. idiots .. the first time I see such a stupid

Reply:
I have appreciated your pull request ! I was about to merge it, I have reviewed the code only and asked nicely about the changes, not beign rude and acting unprofessional to you.

Reply:
If there is duplication, no need of adding extra duplication, you can just leave it cleaner than you found it "The boy scout rule", but instead you prefer insulting 👎